# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1216677329 14400
#      Mon Jul 21 17:55:29 2008 -0400
# Node ID 22a18f2061212e8b99f376900bcfcb03a4d8f652
# Parent  a73b80cd1f10556e1e4641e5174aaf4ad60cfcb4
octave_class::subsasgn: only do internal magic if RHS is not an octave_class object

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,10 +1,13 @@
 2008-07-21  John W. Eaton  <jwe@octave.org>
 
+	* ov-class.cc (octave_class::subsasgn): Only do internal magic if
+	rhs is not an octave_class object.
+
 	* OPERATORS/op-struct.cc: Define concatenation operators for
 	struct/matrix concatenation (valid if matrix is empty).
 	* OPERATORS/op-cell.cc (install_cell_ops): Likewise, for cells.
 
 	* DLD-FUNCTIONS/fltk_backend.cc: Don't include oct.h.
 	Make compilation of entire file conditional on HAVE_FLTK.
 
 	* gl-render.cc: Make compilation of entire file conditional on
diff --git a/src/ov-class.cc b/src/ov-class.cc
--- a/src/ov-class.cc
+++ b/src/ov-class.cc
@@ -314,17 +314,17 @@ octave_class::numeric_conv (const Cell& 
 
 octave_value
 octave_class::subsasgn (const std::string& type,
 			const std::list<octave_value_list>& idx,
 			const octave_value& rhs)
 {
   octave_value retval;
 
-  if (in_class_method ())
+  if (in_class_method () && ! rhs.is_object ())
     {
       // FIXME -- this block of code is the same as the body of
       // octave_struct::subsasgn.  Maybe it could be shared instead of
       // duplicated.
 
       int n = type.length ();
 
       octave_value t_rhs = rhs;
